src/m3ninx/index/segment/fst/segment.go

```diff
@@ -298,16 +297,24 @@ func (r *fsSegment) MatchTerm(field []byte, term []byte) (postings.List, error)
 	return pl, nil
 }
 
-func (r *fsSegment) MatchRegexp(field []byte, regexp []byte, compiled *regexp.Regexp) (postings.List, error) {
+func (r *fsSegment) MatchRegexp(field []byte, regexp []byte, compiled *index.CompiledRegex) (postings.List, error) {
```

Comment:
Why pass `compiled` as a pointer? Since the internal fields are already pointers would it make sense to pass the struct as a value?

Reply:
Sure, will update.